ds3-sdk/src/test/java/com/spectralogic/ds3client/helpers/Ds3ClientHelpers_Test.java

```diff
@@ -242,7 +241,61 @@ private static void checkContents(
         assertThat(contents.getLastModified(), is(lastModified));
         assertThat(contents.getSize(), is(size));
     }
+
     
+    @Test
+    public void testRecoverWriteJob() throws SignatureException, IOException, XmlProcessingException, JobRecoveryException {
+        final Ds3Client ds3Client = buildDs3ClientForBulk();
+
+        final ModifyJobResponse modifyWriteJobResponse = buildModifyWriteJobResponse();
+        Mockito.when(ds3Client.modifyJob(Mockito.any(ModifyJobRequest.class))).thenReturn(modifyWriteJobResponse);
+
+        final Job job = Ds3ClientHelpers.wrap(ds3Client).recoverWriteJob(jobId);
+        //Mockito.verify(ds3Client, Mockito.times(1)).modifyJob(new ModifyJobRequest(jobId));
+        assertThat(job.getJobId(), is(jobId));
+        assertThat(job.getBucketName(), is(MYBUCKET));
+    }
+
+    @Test
+    public void testRecoverWriteJobThrowsJobRecoveryExceptionForWrongRequestType() throws SignatureException, IOException, XmlProcessingException {
+        final Ds3Client ds3Client = buildDs3ClientForBulk();
+
+        final ModifyJobResponse modifyReadJobResponse = buildModifyReadJobResponse();
+        Mockito.when(ds3Client.modifyJob(Mockito.any(ModifyJobRequest.class))).thenReturn(modifyReadJobResponse);
+
+        try {
+            Ds3ClientHelpers.wrap(ds3Client).recoverWriteJob(jobId);
+            Assert.fail("Should have failed with a JobRecoveryException before we got here.");
+        } catch (final JobRecoveryException e) {
+        }
+    }
+
+    @Test
+    public void testRecoverReadJob() throws SignatureException, IOException, XmlProcessingException, JobRecoveryException {
+        final Ds3Client ds3Client = buildDs3ClientForBulk();
+
+        final ModifyJobResponse modifyReadJobResponse = buildModifyReadJobResponse();
+        Mockito.when(ds3Client.modifyJob(Mockito.any(ModifyJobRequest.class))).thenReturn(modifyReadJobResponse);
+
+        final Job job = Ds3ClientHelpers.wrap(ds3Client).recoverReadJob(jobId);
+        assertThat(job.getJobId(), is(jobId));
+        assertThat(job.getBucketName(), is(MYBUCKET));
+    }
+
+    @Test
+    public void testRecoverReadJobThrowsJobRecoveryExceptionForWrongRequestType() throws SignatureException, IOException, XmlProcessingException {
+        final Ds3Client ds3Client = buildDs3ClientForBulk();
+
+        final ModifyJobResponse modifyWriteJobResponse = buildModifyWriteJobResponse();
+        Mockito.when(ds3Client.modifyJob(Mockito.any(ModifyJobRequest.class))).thenReturn(modifyWriteJobResponse);
+
+        try {
+            Ds3ClientHelpers.wrap(ds3Client).recoverReadJob(jobId);
+            Assert.fail("Should have failed with a JobRecoveryException before we got here.");
```

Comment:
You could optionally have the test expect the JobRecoveryException by doing:

`@Test(expect=JobRecoveryException.class)`

and then you could remove the try catch clause.